employees/tests/test_views.py

```diff
@@ -544,3 +545,61 @@ def _assert_response_contain_report(self, response, reports):
                     fields_to_check.append(getattr(report, field))
             for field in fields_to_check:
                 self.assertContains(response, field)
+
+
+class TestAuthorReportProjectView(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.user = UserFactory()
+        self.project1 = ProjectFactory()
+        self.project1.members.add(self.user)
+        current_date = datetime.datetime.now().date()
+        self.report_user = ReportFactory(author=self.user, project=self.project1, date=current_date)
+
+        self.url = reverse(
+            "author-report-project-list",
+            kwargs={
+                "pk": self.user.pk,
+                "project_pk": self.project1.pk,
+                "year": current_date.year,
+                "month": current_date.month,
+            },
+        )
+
+    def test_view_should_display_author_reports_of_project_on_admin_get(self):
+        admin = AdminUserFactory()
+        self.client.force_login(admin)
+        response = self.client.get(self.url)
+        self.assertContains(response, self.report_user.description)
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, AuthorReportProjectView.template_name)
+
+    def test_view_should_display_author_reports_of_project_on_manager_project_get(self):
+        manager = ManagerUserFactory()
+        self.project1.managers.add(manager)
+        self.client.force_login(manager)
+        response = self.client.get(self.url)
+        self.assertContains(response, self.report_user.description)
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, AuthorReportProjectView.template_name)
+
+    def test_view_should_not_display_reports_for_any_other_managers_and_users(self):
```

Comment:
Those should be separate tests, as you actually are checking the manager here (user becomes manager after he is added to project, if not, you are only checking the user here, so please make tests for a manager role).

Reply:
done